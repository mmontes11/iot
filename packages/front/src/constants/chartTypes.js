export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
export const REALTIME = "REALTIME";
