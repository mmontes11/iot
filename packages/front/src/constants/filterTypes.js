export const THING_FILTER_TYPE = "thing";
export const DATE_FILTER_TYPE = "date";
export const FILTER_TYPES = [THING_FILTER_TYPE, DATE_FILTER_TYPE];
