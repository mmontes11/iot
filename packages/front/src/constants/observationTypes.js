export const EVENT_TYPE = "event";
export const MEASUREMENT_TYPE = "measurement";
export const OBSERVATION_TYPES = [EVENT_TYPE, MEASUREMENT_TYPE];
