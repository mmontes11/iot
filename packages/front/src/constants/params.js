export const TYPE = "TYPE";
export const OBSERVATION = "OBSERVATION";
export const GROUPBY = "GROUPBY";
export const THING = "THING";
