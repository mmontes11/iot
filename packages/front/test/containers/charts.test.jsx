import React from "react";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import Charts from "containers/charts";
import { BARCHART, LINECHART } from "constants/chartTypes";
import { initialState, defaultStore, statsItem, eventDataItem, measurementDataItem } from "../constants";

describe("containers/charts", () => {
  it("renders nothing", () => {
    const wrapper = mount(
      <Provider store={defaultStore}>
        <MemoryRouter initialEntries={["/"]} keyLength={0}>
          <Charts charType="foo" />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders when loading", () => {
    const state = {
      ...initialState,
      data: {
        items: [],
        things: [],
        isLoading: true,
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]} keyLength={0}>
          <Charts charType={BARCHART} />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders a bar chart", () => {
    const state = {
      ...initialState,
      data: {
        items: [statsItem],
        things: [],
        isLoading: false,
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]} keyLength={0}>
          <Charts charType={BARCHART} />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders a line chart of event data", () => {
    const state = {
      ...initialState,
      data: {
        items: [eventDataItem],
        things: [],
        isLoading: false,
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]} keyLength={0}>
          <Charts charType={LINECHART} />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders a line chart of measurement data", () => {
    const state = {
      ...initialState,
      data: {
        items: [measurementDataItem],
        things: ["foo", "bar"],
        isLoading: false,
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]} keyLength={0}>
          <Charts charType={LINECHART} />
        </MemoryRouter>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
