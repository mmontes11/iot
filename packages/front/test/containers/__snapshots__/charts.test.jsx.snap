// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/charts renders a bar chart 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Charts)
        charType="BARCHART"
      >
        <Charts
          charType="BARCHART"
          dispatch={[Function]}
          isLoading={false}
          items={
            Array [
              Object {
                "data": Array [
                  Object {
                    "avg": 79,
                    "max": 79,
                    "min": 79,
                    "stdDev": 0,
                    "thing": "arduino",
                  },
                ],
                "type": "temperature",
                "unit": Object {
                  "name": "degrees",
                  "symbol": "°C",
                },
              },
            ]
          }
          things={Array []}
        />
      </Connect(Charts)>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`containers/charts renders a line chart of event data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Charts)
        charType="LINECHART"
      >
        <Charts
          charType="LINECHART"
          dispatch={[Function]}
          isLoading={false}
          items={
            Array [
              Object {
                "items": Array [
                  Object {
                    "phenomenonTime": "2019-04-06T18:59:00.000Z",
                    "values": Array [
                      Object {
                        "thing": "arduino",
                        "value": 1200,
                      },
                    ],
                  },
                ],
                "type": "door-opened",
              },
            ]
          }
          things={Array []}
        />
      </Connect(Charts)>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`containers/charts renders a line chart of measurement data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Charts)
        charType="LINECHART"
      >
        <Charts
          charType="LINECHART"
          dispatch={[Function]}
          isLoading={false}
          items={
            Array [
              Object {
                "items": Array [
                  Object {
                    "phenomenonTime": "2019-04-06T18:59:00.000Z",
                    "values": Array [
                      Object {
                        "thing": "arduino",
                        "value": 79,
                      },
                    ],
                  },
                ],
                "type": "temperature",
                "unit": Object {
                  "name": "degrees",
                  "symbol": "°C",
                },
              },
            ]
          }
          things={
            Array [
              "foo",
              "bar",
            ]
          }
        />
      </Connect(Charts)>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`containers/charts renders nothing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Charts)
        charType="foo"
      >
        <Charts
          charType="foo"
          dispatch={[Function]}
          isLoading={false}
          items={Array []}
          things={Array []}
        />
      </Connect(Charts)>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`containers/charts renders when loading 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Charts)
        charType="BARCHART"
      >
        <Charts
          charType="BARCHART"
          dispatch={[Function]}
          isLoading={true}
          items={Array []}
          things={Array []}
        >
          <Loader>
            <div
              className="is-horizontal-center"
            >
              <figure
                className="image is-128x128"
              >
                <img
                  alt="Loader"
                  src="test-file-stub"
                />
              </figure>
            </div>
          </Loader>
        </Charts>
      </Connect(Charts)>
    </Router>
  </MemoryRouter>
</Provider>
`;
